code-comment-style.md

```diff
@@ -0,0 +1,129 @@
+# Code Comment Style
+
+This document describes the code comment style applied to TiKV repositories. When you are to commit, be sure to follow the style to write good code comments.
+
+## Why a good comment matters?
+
+- To speed up the reviewing process
+- To help maintain the code
+- To improve the API document readability
+- To improve the development efficiency of the whole team
+
+## Where/When to comment?
+
+Write a comment where/when there is context that is missing from the code or is hard to deduce from the code. Specifically, use a comment:
+
+- For important code
+- For obscure code
+- For tricky or interesting code
+- For a complex code block
+- If a bug exists in the code but you cannot fix it or you just want to ignore it for the moment
+- If the code is not optimal but you don’t have a smarter way now
+- To remind yourself or others of missing functionality or upcoming requirements not present in the code
+
+A comment is generally required for:
+
+- Module
+- Type
+- Constant
+- Function
+- Method
+- Variable
+- Typical algorithm
+- Exported name
+- Test case
+- TODO
+- FIXME
+
+## How to comment?
+
+### Format of a good comment (for Rust)
+
+- Non-doc comment
+
+    - Used to document implementation details
+    - Use **//** for a line comment
+    - Use **/* … */** for a block comment (used only when needed)
+    
+- Doc comment
+
+    - Used to document the interface of code (structures, fields, macros, etc.)
+    - Use **///** for a line comment 
+    - Use **//!** for a block comment
+    - Use **rustfmt** to format your code
+
+- Place the single-line and block comment above the code it’s annotating
+- Fold long lines of comments
+- The maximum width for a line is 100 characters
+- Do not use relative URLs; use **absolute URLs** instead

```

Comment:
It is absolutely appropriate to use relative URLs in autogenerated documentation. The user's machine will generate them via `cargo doc` and the hosting location may be anything. So using an absolute URL will either break local documentation, or hosted documentation.

For example, this is how you link to another struct: https://github.com/tikv/client-rust/blob/master/src/raw.rs#L75

Reply:
As confirmed with @BusyJay, we should use relative URLs for local links.  Now updated as: Use "Use relative URLs when you link to positions on your local machine"